src/pyfmi/fmi.pyx

```diff
@@ -8222,12 +8261,49 @@ cdef class FMUModel(FMUModelME1):
     def __init__(self, fmu, path='.', enable_logging=True):
         print("WARNING: This class is deprecated and has been superseded with FMUModelME1. The recommended entry-point for loading an FMU is now the function load_fmu.")
         FMUModelME1.__init__(self,fmu,path,enable_logging)
-        
+
 def _handle_load_fmu_exception(fmu, log_data):
     for log in log_data:
         print(log)
 
-def load_fmu(fmu, path = '.', enable_logging = None, log_file_name = "", kind = 'auto', log_level=FMI_DEFAULT_LOG_LEVEL):
+def display_path_deprecation(path):
+    if path != '.':
+        w = "The argument 'path' is deprecated and will be removed in a future version."
+        w += " Please specify a full path to fmu via the argument 'fmu' instead."
+        logging.warning(w)
+
+def check_fmu_args(allow_unzipped_fmu, fmu, fmu_full_path):
+    """ Function utilized by two base classes and load_fmu that does the
+        error checking for the three input arguments named 'allow_unzipped_fmu', 'fmu' and
+        the constructed variable 'fmu_full_path'.
+    """
+    if allow_unzipped_fmu:
+        if not os.path.isdir(fmu):
+            msg = "Argument named 'fmu' must be a directory if argument 'allow_unzipped_fmu' is set to True."
+            raise FMUException(msg)
+        if not os.path.isfile(os.path.join(fmu, 'modelDescription.xml')):
+            err_msg = "Specified fmu path '{}' needs".format(fmu)
+            err_msg += " to contain a modelDescription.xml according to the FMI specification."
+            raise FMUException(err_msg)
+    else:
+        # Check that the file referenced by fmu is appropriate
+        if not fmu_full_path.endswith('.fmu' if isinstance(fmu_full_path, str) else encode('.fmu')):
+            raise FMUException("FMUModel must be instantiated with an FMU (.fmu) file.")
+
+        if not os.path.isfile(fmu_full_path):
+            raise FMUException('Could not locate the FMU in the specified directory.')
+
+cdef FMIL.fmi_version_enu_t import_and_get_version(FMIL.fmi_import_context_t* context, char* fmu_full_path, char* fmu_temp_dir, int allow_unzipped_fmu):
```

Comment:
I believe we should remove this method and directly invoke fmi_import_and_get_version. The only difference here is that we send in NULL in one branch. This can be handled directly at the places where this class is invoked.

Reply:
Hi, I agree with you, but my reasoning for this is also to reduce code duplication. I guess we have 2 options, using a function or duplicate the code in 5 different locations. I tried to do inline conditionals, i.e. something like:
`f('foo' if conditional else 'bar', ...)` but I didn't get it to work as Cython complained about the type of the temporary variable created within the expression.
Personally, I think we gain more by using this function instead of adding the following expression in 5 different locations:
`if allow_unzipped_fmu:
    f(..., NULL, ...)
else:
    f(..., fmu_full_path, ...)
`
What are your thoughts? In general I think there is a lot of code duplication in pyfmi, and I believe working to fix that would make it easier to work with.

Reply:
What about changing fmu_full_path to a C type and initialize it to NULL? Then if we use it, we set it to the full path, otherwise it is NULL and this call is identical in both cases. I.e.: f(..., fmu_full_path,...)

Reply:
Yes you're right that would make that line identical in all places but then we would have to add logic instead to check if it is used or not and I don't think we gain anything. For example, assume we initialize it to NULL, then I think (correct me if I'm wrong) we need to add code to see if we should set it to os.path.abspath(os.path.join(path,fmu)), that is, if allow_unzipped_fmu is set to True (which adds a new if-block) in all locations where this variable is initialized.

Reply:
Hmm ok, I'd still rather have it as something fmi_import_get_fmi_version(..., NULL if allowed else fmu_path,...) and remove this method.